catboost/libs/documents_importance/docs_importance_helpers.cpp

```diff
@@ -21,16 +23,29 @@ TVector<TVector<double>> TDocumentImportancesEvaluator::GetDocumentImportances(c
     localExecutor.ExecRange([&] (int treeId) {
         leafIndices[treeId] = BuildIndicesForBinTree(Model, binarizedFeatures, treeId);
     }, NPar::TLocalExecutor::TExecRangeParams(0, TreeCount), NPar::TLocalExecutor::WAIT_COMPLETE);
-
     UpdateFinalFirstDerivatives(leafIndices, pool);
-    TVector<TVector<double>> documentImportances(DocCount, TVector<double>(pool.Docs.GetDocCount()));
 
-    localExecutor.ExecRange([&] (int docId) {
-        // The derivative of leaf values with respect to train doc weight.
-        TVector<TVector<TVector<double>>> leafDerivatives(TreeCount, TVector<TVector<double>>(LeavesEstimationIterations)); // [treeCount][LeavesEstimationIterationsCount][leafCount]
-        UpdateLeavesDerivatives(docId, &leafDerivatives);
-        GetDocumentImportancesForOneTrainDoc(leafDerivatives, leafIndices, &documentImportances[docId]);
-    }, NPar::TLocalExecutor::TExecRangeParams(0, DocCount), NPar::TLocalExecutor::WAIT_COMPLETE);
+    TVector<TVector<double>> documentImportances(DocCount, TVector<double>(pool.Docs.GetDocCount()));
+    const ui32 docDivider = 1000;
+    size_t docBlockSize = DocCount < docDivider ? CB_THREAD_LIMIT : CB_THREAD_LIMIT * (CeilDiv(DocCount, docDivider));
+    TFstrLogger documentsLogger(DocCount, "documents processed", "Processing documents...", 1);
+    TProfileInfo processDocumentsProfile(DocCount);
+
+    for (size_t start = 0; start < DocCount; start += docBlockSize) {
+        size_t end = Min(start + docBlockSize, static_cast<size_t>(DocCount));
```

Comment:
1. Please use Min<size_t>(...), and remove static cast.
2. Please declare end with const.